### PR TITLE
Issue #2461 Genric: Add error msg in case required option not provided

### PR DESCRIPTION
**Pull request by praveenkumar**
@agajdosi Have a look.

### CODE REVIEW

**Comment by agajdosi** [comment]
@praveenkumar That's exactly what I imagined. Tried it and looks fine, I am however in hurry now, so I can't fully check and approve.

File: cmd/minishift/cmd/start.go [hunk 1]
```diff
@@ -161,6 +161,17 @@ func runStart(cmd *cobra.Command, args []string) {
...
+	if viper.GetString(configCmd.VmDriver.Name) == genericDriver {
+		if viper.GetString(configCmd.RemoteIPAddress.Name) == "" ||
+			viper.GetString(configCmd.RemoteSSHUser.Name) == "" ||
+			viper.GetString(configCmd.SSHKeyToConnectRemote.Name) == "" {
+			msg := fmt.Sprintf("Generic driver require additional information about IP address of remote machine, ssh key location and ssh username.\n"+
+				"Please enable experimental features of Minisift and provide following flags to use generic driver:\n"+
+				"--%s string\n--%s string\n--%s string\n", configCmd.RemoteIPAddress.Name, configCmd.RemoteSSHUser.Name, configCmd.SSHKeyToConnectRemote.Name)
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error: %s", msg))
+		}
+	}
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: @praveenkumar  can we create an function for this? IMO we should keep `runStart` as clean  and easy to read as much as possible. 

I suggest the function name as `validateGenricDriverFlags()`

File: cmd/minishift/cmd/start.go [hunk 2]
```diff
@@ -161,6 +161,17 @@ func runStart(cmd *cobra.Command, args []string) {
 	libMachineClient := libmachine.NewClient(state.InstanceDirs.Home, state.InstanceDirs.Certs)
 	defer libMachineClient.Close()
 
+	if viper.GetString(configCmd.VmDriver.Name) == genericDriver {
+		if viper.GetString(configCmd.RemoteIPAddress.Name) == "" ||
+			viper.GetString(configCmd.RemoteSSHUser.Name) == "" ||
+			viper.GetString(configCmd.SSHKeyToConnectRemote.Name) == "" {
+			msg := fmt.Sprintf("Generic driver require additional information about IP address of remote machine, ssh key location and ssh username.\n"+
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: `Generic driver require additional information i.e. IP address of remote machine, path to ssh key and ssh username of the remote host.

File: cmd/minishift/cmd/start.go [hunk 3]
```diff
@@ -161,6 +161,17 @@ func runStart(cmd *cobra.Command, args []string) {
...
+				"Please enable experimental features of Minisift and provide following flags to use generic driver:\n"+
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start.go]: `Enable experimental features of Minisift and provide following flags to use generic driver`

**Pull request merged**